Canine.py

```diff
@@ -3,8 +3,9 @@
 
 class Canine(Animal, metaclass=ABCMeta):
     @abstractmethod
-    def __init__(self, name):
+    def __init__(self, name, makeNoise=None):
         super().__init__(name)
 
```

Comment:
Set "makeNoise" as a private attribute in constructor
Like:
`self.__noiseAlgo = makeNoise
`